lib/ansible/plugins/action/debug.py

```diff
@@ -61,8 +59,10 @@ def run(self, tmp=None, task_vars=None):
                             raise AnsibleUndefinedVariable
                         # If var name is same as result, try to template it
                         results = self._templar.template("{{" + results + "}}", convert_bare=True, fail_on_undefined=True)
-                except AnsibleUndefinedVariable:
+                except AnsibleUndefinedVariable as e:
                     results = "VARIABLE IS NOT DEFINED!"
+                    if self._display.verbosity > 0:
+                        results += ": %s" % to_text(e)
```

Comment:
As sivel pointed out on IRC, it would be good if all of the string literals here were defined as unicode strings.  Like this:
``` python

results = u"VARIABLE IS NOT DEFINED"
if self._display.verbosity > 0:
    results += u": %s" % to_text(e)
```